src/renderer/wg_engine/tvgWgGeometry.cpp

```diff
@@ -328,24 +354,30 @@ void WgGeometryDataGroup::stroke(const RenderShape& rshape)
 {
     assert(rshape.stroke);
     if (rshape.stroke->dashPattern) {
-        // first step: decode path data
+        // decode path data
         WgGeometryDataGroup segments{};
         decodePath(rshape, &segments);
-        // second step: split path to segments using dash patterns
+        // trim segments
+        WgGeometryDataGroup trimmed{};
+        trimOutlines(&segments, &trimmed, rshape.stroke->trim.begin, rshape.stroke->trim.end);
```

Comment:
I think you can skip the trimOutlines when the trim region is default (begin 0.0, end = 1.0) ?